lib/streamlit/elements/widgets/slider.py

```diff
@@ -166,14 +194,171 @@ def serialize(self, v: Any) -> list[Any]:
 
 
 class SliderMixin:
-    @gather_metrics("slider")
+    # For easier readability, all the arguments with un-changing types across these overload signatures have been
+    # collapsed onto a single line.
+
+    # fmt: off
+    # If min-value is provided and a numeric type, and value (if provided) is a singular numeric, return the same numeric type.
+    @overload
+    def slider(
+        self,
+        label: str,
+        min_value: SliderNumericT,
+        max_value: SliderNumericT | None = None,
+        value: SliderNumericT | None = None,
+        step: StepNumericT[SliderNumericT] | None = None,
+        format: str | None = None, key: Key | None = None, help: str | None = None, on_change: WidgetCallback | None = None, args: WidgetArgs | None = None, kwargs: WidgetKwargs | None = None, *, disabled: bool = False, label_visibility: LabelVisibility = "visible"
+    ) -> SliderNumericT:
+        ...
+
+    # If max-value is provided and a numeric type, and value (if provided) is a singular numeric, return the same numeric type.
+    @overload
+    def slider(
+        self,
+        label: str,
+        min_value: SliderNumericT | None = None,
+        *,
```

Comment:
Here, we actually don't need it. But we do in some of the other signatures whenever an argument has a "default value" that depends on the type of other values. For example, 
```python
   @overload
    def slider(
        self,
        label: str,
        min_value: SliderNumericT | None = None,
        max_value: SliderNumericT | None = None,
        *,
        value: SliderNumericSpanT[SliderNumericT],
        step: StepNumericT[SliderNumericT] | None = None,
        format: str | None = None, key: Key | None = None, help: str | None = None, on_change: WidgetCallback | None = None, args: WidgetArgs | None = None, kwargs: WidgetKwargs | None = None, disabled: bool = False, label_visibility: LabelVisibility = "visible"
    ) -> tuple[SliderNumericT, SliderNumericT]:
        ...
```

Must have value as a keyword argument to handle the case where the user calls `slider(label="foo", value=[0, 10])` returning a tuple of values. I need the `*` to because `value` follows `min_value` and `max_value` which both have default arguments here, but `value` does not have a default argument because it's default argument is actually 0, which is not a `SliderNumericSpanT` type.